streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -202,18 +206,21 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
 
   /** Allocate all unallocated blocks to the given batch. */
   def allocateBlocksToBatch(batchTime: Time): Unit = {
+    require(receivedBlockTracker != null, "ReceiverTracker should be started first.")
```

Comment:
Per comments above, I don't know if you need to assert about the state here. I suppose you'll already get an error if this isn't true, but, this would indicate something else already went wrong.